src/ocsp.c

```diff
@@ -616,50 +623,60 @@ WOLFSSL_OCSP_CERTID* wolfSSL_OCSP_cert_to_id(
                 issuer->derCert->length);
         ret = AddCA(cm, &derCert, WOLFSSL_USER_CA, 1);
         if (ret != WOLFSSL_SUCCESS) {
-            wolfSSL_CertManagerFree(cm);
-            return NULL;
+            goto out;
         }
     }
 
+    ret = -1;
```

Comment:
Use `cm->heap` hint for the XMALLOC's and XFREE's below.

Reply:
done